tests/fips/mozilla_nss/firefox_nss.pm

```diff
@@ -80,36 +79,52 @@ sub run {
     assert_screen "firefox-confirm-fips_enabled";
     send_key "esc";    # Quit device manager
 
+    # Close Firefox
     quit_firefox;
+    wait_still_screen 10;
     assert_screen "generic-desktop";
+    wait_still_screen 10;
+
 
     # "start_firefox" will be not used, since the master password is
     # required when firefox launching in FIPS mode
-    x11_start_program('firefox --setDefaultBrowser https://html5test.opensuse.org', target_match => 'firefox-fips-password-inputfiled', match_timeout => 360);
-
+    # x11_start_program('firefox --setDefaultBrowser https://html5test.opensuse.org', target_match => 'firefox-fips-password-inputfiled', match_timeout => 180);
+    # Adjust the 2nd Firefox launch via xterm as workaround to avoid the race condition during password interaction while internet connection busy
+    x11_start_program('xterm');
+    mouse_hide(1);
+    type_string("firefox --setDefaultBrowser https://html5test.opensuse.org\n");
+    wait_still_screen 120;
```

Comment:
Maybe you can use assert_screen $timeout instead the wait_still_screen, I don't think it is suitable here.

Reply:
thanks for the comment, I will change it.